### PR TITLE
Feature/donations function update

### PR DESCRIPTION
**Pull request by VladimirKarin**
Updated:

1. donation func. in businessRules fold:

- getTotalSumDonatedForIdea updated.
- getDonationsForIdea created. 
- getUsersDonations created.

2. validateUserId func. in validatation fold.

- validateUserid func. update.

### CODE REVIEW

File: businessRules/donations.js [hunk 1]
```diff
@@ -55,10 +56,32 @@ function createDonationByRegisteredUser(ideaId, userId, sum) {
...
+    return {
+        numberOfDonations: idea.totalDonationSum,
+    };
+}
+
+function getDonationsForIdea(ideaId) {
+    validateIdea(ideaId);
+    const donations = getDonations();
+    const donationsForIdea = donations.filter(
+        (donation) => (donation.ideaId = ideaId)
```
Comment (fedosejev) [inline businessRules/donations.js:68]: Can you spot the error here?

File: businessRules/donations.js [hunk 2]
```diff
@@ -55,10 +56,32 @@ function createDonationByRegisteredUser(ideaId, userId, sum) {
...
+    );
+    return donationsForIdea;
+}
+
+function getUsersDonations(userId) {
+    validateUserId(userId);
+    const donations = getDonations();
+    const usersDonations = donations.filter(
+        (donation) => (donation.userId = userId)
```
Comment (fedosejev) [inline businessRules/donations.js:77]: Can you spot the error here?

File: businessRules/donations.js [hunk 3]
```diff
@@ -55,10 +56,32 @@ function createDonationByRegisteredUser(ideaId, userId, sum) {
 function getTotalSumDonatedForIdea(ideaId) {
     const idea = getIdea(ideaId);
 
-    return idea.totalDonationSum;
+    return {
+        numberOfDonations: idea.totalDonationSum,
+    };
```
Comment (fedosejev) [inline businessRules/donations.js:61]: The function is called `getTotalSumDonatedForIdea`, but you're returning `numberOfDonations`?

File: businessRules/donations.js (hunk 1 above)
Comment (VladimirKarin) [reply businessRules/donations.js]: ideaId == donation.ideaId ?

File: businessRules/donations.js (hunk 2 above)
Comment (VladimirKarin) [reply businessRules/donations.js]: Same as previous?

File: businessRules/donations.js (hunk 3 above)
Comment (VladimirKarin) [reply businessRules/donations.js:61]: Fixed.

File: businessRules/donations.js [hunk 4]
```diff
@@ -55,10 +56,32 @@ function createDonationByRegisteredUser(ideaId, userId, sum) {
...
+    return {
+        totalSumDonatedForIdea: idea.totalDonationSum,
+    };
+}
+
+function getDonationsForIdea(ideaId) {
+    validateIdea(ideaId);
+    const donations = getDonations();
+    const donationsForIdea = donations.filter(
+        (donation) => ideaId == donation.ideaId
```
Comment (fedosejev) [inline businessRules/donations.js:68]: Why `==` and not `===`?

File: businessRules/donations.js [hunk 5]
```diff
@@ -55,10 +56,32 @@ function createDonationByRegisteredUser(ideaId, userId, sum) {
...
+    );
+    return donationsForIdea;
+}
+
+function getUsersDonations(userId) {
+    validateUserId(userId);
+    const donations = getDonations();
+    const usersDonations = donations.filter(
+        (donation) => userId == donation.userId
```
Comment (fedosejev) [inline businessRules/donations.js:77]: Why `==` and not `===`?

File: businessRules/donations.js (hunk 4 above)
Comment (VladimirKarin) [reply businessRules/donations.js]: My bad.

File: businessRules/donations.js (hunk 5 above)
Comment (VladimirKarin) [reply businessRules/donations.js]: Fixed

**Review by fedosejev — APPROVED** [review]

**Pull request merged**